documentation/decisions/adr/2023-12-15-deliverable-reporting-strategy.md

```diff
@@ -0,0 +1,371 @@
+- If necessary, users should be able to change the reporting logic without rewriting significant sections of the source code
+
+## Options considered
+
+### Assigning issues to deliverables
+
+- **Milestones per 10k:** Add the issue to a milestone that represents a 10k deliverable and tag the 30k deliverable in the body of the milestone
+- **Milestones per 30k:** Add the issue to a milestone that represents a 30k deliverable
+- **Labels:** Tag the issue with a label that represents a 30k deliverable
+- **Deliverable column:** Add a custom "deliverable" column to the GitHub projects with values representing each 30k deliverable
+- **Reserved phrase:** Use a reserved phrase to tag a 30k deliverable in the body of the issue
+
+### Limiting the 30k deliverables included in reporting
+
+- **Label:** Use a label to indicate when a deliverable should be included in reporting
+- **Status:** Use the status of the deliverable in the provisional roadmap project
+- **GitHub action:** Explicitly list the deliverables to include in the GitHub action for reporting
+
+## Decision outcomes <!-- REQUIRED -->
+
+### Assigning issues to deliverables
+
+We've decided to use a "deliverable" column to assign issues to a parent 30k deliverable. This involves:
+
+- creating a single select column in both the product roadmap and the sprint board
+- populating that column with options that represent each 30k deliverables in our roadmap
+- using that column to indicate the 30k deliverable that an issue is assigned to
+
+- **Positive outcomes**
+  - We can filter, group, and sort issues by deliverable in GitHub projects and GitHub's automated insight reporting.
+  - The engineering team can continue to use GitHub milestones to organize issues into units of work that are smaller than a 30k deliverable. **Note:** While enabling the engineering team to use milestones to organize issues into smaller units of work, we want to avoid situations in which a given milestone contains issues from multiple 30k deliverables -- we can enforce this with a linter as well as a view in the GitHub project.
+  - The logic for grouping issues by deliverable will be consistent between GitHub insight reports and our custom-built reports.
+- **Negative outcomes**
+  - We won't be able to filter for issues assigned to a given 30k deliverable within the GitHub repository.
+  - Updating a field on a project can be done from the issue page itself, but the issue first needs to be added to the project before the field is updated. This requires an extra step compared to assigning a label or milestone.
+  - We'll have to make sure the list of options in the deliverable column is consistent across GitHub projects in order to join issues from different projects correctly in our custom reporting.
+  - We'll have to update the current logic in our custom reporting so that issues are joined to their parent deliverable using the value of this column.
+
```

Comment:
```suggestion
- When creating a new Issue, it's not sufficient to add that Issue to a Milestone; it must also be added to the 30k deliverable for that Milestone for reporting purposes. We will need to monitor for Issues that do not have a 30k and triage them regularly. 

```

Reply:
I'm still wondering if we might make it easier to associate Issues with the appropriate Milestone and 30k deliverable. One idea that comes to mind is prefixing all Milestone titles with an ID that matches the ID of the deliverable — e.g.: 
- Issue #n has the title "[30k ]: My Deliverable Name" 
- The deliverable column uses that Issue's ID in its title, e.g. "[#n]: My Deliverable Name"
- Milestones that address that deliverable use that same ID in their titles, e.g. "[#n]: My Milestone Name" 
- Then, when creating an Issue, it is clear that you should choose a Milestone and deliverable that are prefixed with same ID. This may make it easier to choose the right deliverable when it is not apparent which 30k a Milestone addresses, or for individual contributors who are closer to the ground, more concerned with Milestones, and unsure with which 30k an individual ticket addresses.

Reply:
> One idea that comes to mind is prefixing all Milestone titles with an ID that matches the ID of the deliverable

I like this idea as a way to make the intended deliverable clear, though I worry it could introduce discrepancies if an issue gets reassigned to a different deliverable but forgets to update the title of the issue as well. 

Part of the reason we created this ADR was to limit the "sources of truth" that associate an issue to a deliverable.